pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2190,8 +2190,10 @@ public CompletableFuture<Void> checkTopicNsOwnership(final String topic) {
     }
 
     public CompletableFuture<Integer> unloadServiceUnit(NamespaceBundle serviceUnit,
```

Comment:
Should we add a new method for this public method change instead of adding a new parameter?

Reply:
Probably not. We need to pass this flag to topic.close(..) in this func anyway and reduce some code redundancy.